src/lang-promql/parser/parser.ts

```diff
@@ -273,6 +264,45 @@ export class Parser {
     }
   }
 
+  private checkVectorSelector(node: Subtree): void {
+    const labelMatchers = buildLabelMatchers(
+      retrieveAllRecursiveNodes(walkThrough(node, LabelMatchers, LabelMatchList), LabelMatchList, LabelMatcher),
+      this.state
+    );
+    // A Vector selector must contain at least one non-empty matcher to prevent
+    // implicit selection of all metrics (e.g. by a typo).
+    let empty = true;
+    let i = 0;
+    while (i < labelMatchers.length && empty) {
+      empty = labelMatchers[i].matchesEmpty();
+      i++;
+    }
+    if (labelMatchers.length > 0 && empty) {
```

Comment:
Once you include the metric name in the label matchers, I don't think it can ever happen that `labelMatchers.length` is 0, as then you would not end up in this code.